src/Utilities/PredicateBuilder.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+using ePiggyWeb.DataBase.Models;
+
+namespace ePiggyWeb.Utilities
+{
+    public static class PredicateBuilder
+    {
+        public static Expression<Func<IEntryModel, bool>> BuildEntryFilter(IEnumerable<int> idEnumerable, int userId)
+        {
+            if (idEnumerable is null)
+            {
+                return x => x.UserId == userId;
+            }
+
+            var idArray = idEnumerable as int[] ?? idEnumerable.ToArray();
+
+            if (!idArray.Any())
+            {
+                return x => x.UserId == userId;
+            }
+
+            Expression<Func<IEntryModel, bool>> filter = x => x.Id == idArray[0] && x.UserId == userId;
```

Comment:
isn't that a lambda expression ?

Reply:
But it does not fulfill the requirement?

Reply:
Now that you say it, maybe it should actually. We need to ask the lecturer